0017-chart-namespace-overrides/README.md

```diff
@@ -0,0 +1,413 @@
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+      - [Unit tests](#unit-tests)
+      - [e2e tests](#e2e-tests)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Summary
+
+This ZEP proposes to enable a namespace override for charts similar to the namespace override [functionality available in UDS CLI](https://uds.defenseunicorns.com/reference/bundles/overrides/#namespace).  This would allow namespaces of charts within a Zarf package to be overridden so that multiples of the same Zarf package could be deployed to the same cluster under different namespaces (without needing to maintain variants of the same package).
+
+## Motivation
+
+Doing this allows more flexibility with certain Zarf packages where you may want to have multiples of them installed in the cluster with slightly different configurations (such as [GitLab Runners](https://github.com/defenseunicorns/uds-package-gitlab-runner)).  Right now the release namespace of any chart has to be hardcoded into the package and will be overwritten even if the chart allows namespace overrides for some manifests within the chart.  The current behavior is also different from what Helm does by default which may not be what users of Zarf expect (Helm allows the use of the `namespace` flag on install to set the Chart's namespace without it needing to be baked into the Chart).
+
+### Goals
+
+- Provide a way for an already created Zarf package containing Helm Charts to be easily installed more than once with different configurations
+
+### Non-Goals
+
+- Move away from the declarative nature of Zarf packages
+
+## Proposal
+
+The proposed solution is to introduce a new named override config to Zarf to allow for a managed way to provide overrides for namespaces and eventually different values.  This allows for potential future override expansion while also forcing the overrides to be named and versioned to a package rather than be as fluid as an existing zarf-config file helping reduce declarative loss. These overrides could also eventually be signed as an artifact if desired.
+
+### User Stories (Optional)
+
+#### Story 1
+
+**As** Jacquline **I want** to be able to set namespace overrides **so that** I can install the same package with different configurations in different namespaces.
+
+**Given** I have a Zarf Package with a chart named `my-chart` in a component named `my-component`
+**And** I have a new ZarfOverrideConfig created from the following
+```yaml
+kind: ZarfOverrideConfig
+metadata:
+  name: test-override
+  ref: oci://my-registry/test:0.1.0
+  version: 0.1.0
+
+overrides:
```

Comment:
I'd expect at least a rough schema of this new config described in the design details section. My biggest concern is two folded, how many fields do we want to allow to override? At which point we'd say that overrides are too big and you should consider modifying your package?

Reply:
Yeah to add to that, I'm worried about the UX for someone looking to pick up overrides. How can we make it intuitive for users to know which fields they can override? How can a user validate that their overrides are working as expected before a deploy?

Reply:
changed the proposed solution though this is still an issue in the current proposal - IMO we need a clear enough cut (i.e. fields that affect the deployment) and clear documentation on that - we also should fail fast when the user puts something somewhere they shouldn't (since this is all just in the one zarf.yaml)